### DISCUSSION TITLE
File matched by custom fileMatch but not taken as package file

### DISCUSSION THREAD

**Discussion by YinleXu** [renovatebot/renovate#29800] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Gitlab, renovate version 37.408.0

### Please tell us more about your question or problem

I have an ansible galaxy dependency file with name "_requirements_galaxy.yml_". This does not match the default pattern
```
"(^|/)(galaxy|requirements)(\\.ansible)?\\.ya?ml$"
``` 
Therefore I have added fileMatch config in the repo's renovate.json config file
```
"ansible-galaxy":{
    "fileMatch": ["(^|/)requirements_galaxy.ya?ml$"],
    "enabled": true
}
```
With this config, I can see the file being matched from the debug log but it seems it is not being treated as a package file. Therefore, no MR/PR was created for the dependencies in the file. See more details in the logs.

I have recreated the same issue with this [Github repo](https://github.com/YinleXu/renovate_test). You can see the [galaxy.ansible.yml](https://github.com/YinleXu/renovate_test/blob/main/galaxy.ansible.yml) file, which is matched by the default pattern, gets PRs created while [requirements_galaxy.yml](https://github.com/YinleXu/renovate_test/blob/main/requirements_galaxy.yml) with same content did not. Although the file was actually matched evidently from the debug log.

Am I missing any config in addition to "fileMatch"?

### Logs (if relevant)

<details><summary>Logs</summary>

```

DEBUG: Matched 2 file(s) for manager ansible-galaxy: galaxy.ansible.yml, requirements_galaxy.yml (repository=YinleXu/renovate_test)
DEBUG: Matched 1 file(s) for manager pip_requirements: requirements.txt (repository=YinleXu/renovate_test)
DEBUG: manager extract durations (ms) (repository=YinleXu/renovate_test)
       "managers": {"pip_requirements": 2, "ansible-galaxy": 2}
DEBUG: Found ansible-galaxy package files (repository=YinleXu/renovate_test)
DEBUG: Found pip_requirements package files (repository=YinleXu/renovate_test)
DEBUG: Found 2 package file(s) (repository=YinleXu/renovate_test)
 INFO: Dependency extraction complete (repository=YinleXu/renovate_test, baseBranch=main)
       "stats": {
         "managers": {
           "ansible-galaxy": {"fileCount": 1, "depCount": 4},
           "pip_requirements": {"fileCount": 1, "depCount": 2}
         },
         "total": {"fileCount": 2, "depCount": 6}
```

</details>

**Comment by rarkins** [comment]
Most likely the file you're matching does not contain the same file structure which renovate is expecting for ansible galaxy

**Reply by YinleXu** [reply]
hi @rarkins , thanks for replying. However, the two files have exactly the same content, one being copy of the other. However, galaxy.ansible.yml (default matching) got PRs created while requirements_galaxy.yml did not.

**Reply by rarkins** [reply]
It seems that the ansible-galaxy manager has a check for file name patterns within it too: https://github.com/renovatebot/renovate/blob/1c3dbca90ede3d6230e1a0fecc20798572065cc3/lib/modules/manager/ansible-galaxy/extract.ts#L30-L38

@secustor could this be replaced with a check on _content_ instead of file name?

**Reply by secustor** [reply]
With the support for schemas while parsing YAML that should be possible now.

**Reply by rarkins** [reply, marked as answer]
Converted to issue: https://github.com/renovatebot/renovate/issues/29816
